yt/data_objects/static_output.py

```diff
@@ -808,6 +808,22 @@ def _setup_particle_types(self, ptypes=None):
     _last_finfo = None
 
     def _get_field_info(self, ftype, fname=None):
+        field_info, ambiguous = self._get_field_info_helper(ftype, fname)
+
+        if ambiguous:
+            ft, fn = field_info.name
+            msg = (
+                f"The requested field name '{fn}' "
+                "is ambiguous and corresponds to any one of "
+                f"the following field types:\n {self.field_info._ambiguous_field_names[fn]}\n"
+                "Please specify the requested field as an explicit "
+                "tuple (ftype, fname).\n"
+                f"Defaulting to '({ft}, {fn})'."
```

Comment:
so the suggestion can be copy-pasted back into the culprit code
```suggestion
                f"Defaulting to ('{ft}', '{fn}')."
```

Reply:
the way you implemented it now works too